src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -452,6 +452,23 @@ const JupyterFrameManager = ({ onClose, frameRef }) => {
   return null
 }
 
+const PeriodicCookieSetter = ({ namespace, clusterName }) => {
+  const signal = useCancellation()
+
+  const periodicallySetCookie = async () => {
+    while (!signal.aborted) {
+      withErrorIgnoring(() => Ajax().Jupyter.notebooks(namespace, clusterName).setCookie())()
```

Comment:
One last thought. Not critical, but this can be cancellable.
```suggestion
      withErrorIgnoring(() => Ajax(signal).Jupyter.notebooks(namespace, clusterName).setCookie())()
```

Reply:
Since `setCookie` is a post, I thought it might be better to keep it from being cancellable